### PR TITLE
Rework thread stopping to avoid double lock

### PR DESCRIPTION
**Pull request by anosach-luxoft**

### CODE REVIEW

File: src/components/utils/src/threads/posix_thread.cc [hunk 1]
```diff
@@ -58,7 +56,12 @@ namespace threads {
 CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
 
 void sleep(uint32_t ms) {
+#ifdef SDL_CPP11
```
Comment (pestOO) [inline src/components/utils/src/threads/posix_thread.cc:59]: May be `#if __cplusplus >= 201103L`?

Comment (malirod) [inline src/components/utils/src/threads/posix_thread.cc:59]: @pestOO, exactly this is hidden behind SDL_CPP11 in macro.h

**Comment by malirod** [comment]
@pestOO, please, review. I am ready to merge.

File: src/components/utils/src/threads/posix_thread.cc (hunk 1 above)
Comment (pestOO) [inline src/components/utils/src/threads/posix_thread.cc:59]: Great)

**Comment by pestOO** [comment]
Reviewed

**Pull request merged**